pkgs/applications/audio/zynaddsubfx/default.nix

```
@@ -1,30 +1,118 @@
-{ stdenv, fetchurl, alsaLib, cairo, cmake, libjack2, fftw, fltk13, lash,  libjpeg
-, libXpm, minixml, ntk, pkgconfig, zlib, liblo
+{ stdenv
+, lib
+, fetchFromGitHub
+, callPackage
+
+  # Required build tools
+, cmake
+, makeWrapper
+, pkgconfig
+
+  # Required dependencies
+, fftw
+, liblo
+, minixml
+, zlib
+
+  # Optional dependencies
+, alsaSupport ? true
+, alsaLib ? null
+, dssiSupport ? false
+, dssi ? null
+, ladspaH ? null
+, jackSupport ? true
+, libjack2 ? null
+, lashSupport ? false
+, lash ? null
+, ossSupport ? true
+, portaudioSupport ? true
+, portaudio ? null
+
+  # Optional GUI dependencies
+, guiModule ? "off"
+, cairo ? null
+, fltk13 ? null
+, libGL ? null
+, libjpeg ? null
+, libX11 ? null
+, libXpm ? null
+, ntk ? null
+
+  # Test dependencies
+, cxxtest
 }:
 
-stdenv.mkDerivation  rec {
+assert alsaSupport -> alsaLib != null;
+assert dssiSupport -> dssi != null && ladspaH != null;
+assert jackSupport -> libjack2 != null;
+assert lashSupport -> lash != null;
+assert portaudioSupport -> portaudio != null;
+
+assert builtins.any (g: guiModule == g) [ "fltk" "ntk" "zest" "off" ];
+assert guiModule == "fltk" -> fltk13 != null && libjpeg != null && libXpm != null;
+assert guiModule == "ntk" -> ntk != null && cairo != null && libXpm != null;
+assert guiModule == "zest" -> libGL != null && libX11 != null;
+
+let
+  mruby-zest = callPackage ./mruby-zest.nix { };
+in stdenv.mkDerivation rec {
   pname = "zynaddsubfx";
   version = "3.0.5";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/zynaddsubfx/zynaddsubfx-${version}.tar.bz2";
-    sha256 = "0qwzg14h043rmyf9jqdylxhyfy4sl0vsr0gjql51wjhid0i34ivl";
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "1vh1gszgjxwn8m32rk5222z1j2cnjax0bqpag7b47v6i36p2q4x8";
+    fetchSubmodules = true;
   };
 
-  buildInputs = [ alsaLib cairo libjack2 fftw fltk13 lash libjpeg libXpm minixml ntk zlib liblo ];
-  nativeBuildInputs = [ cmake pkgconfig ];
-
   patchPhase = ''
     substituteInPlace src/Misc/Config.cpp --replace /usr $out
   '';
 
-  hardeningDisable = [ "format" ];
+  nativeBuildInputs = [ cmake makeWrapper pkgconfig ];
```

```suggestion
  nativeBuildInputs = [ cmake makeWrapper pkg-config ];
```

Also rename in inputs.